frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  CommonTemplatesValidation,
+  IntegerValidationObj,
+  IntegerValidtionType,
+} from './validation-types';
+import {
+  getRelevantTemplates,
+  getValidationsFromTemplates,
+  getFieldValidations,
+  getValidationAttributes,
+  getBetweenIntegerValidationObj,
+  getMinIntegerValidationObj,
+  getMaxIntegerValidationObj,
+} from './selectors';
+
+// TODO: Add all the fields in the form
+// For each field we need to check for validations
+const IDToJsonPath = {
+  [getFieldId(VMSettingsField.MEMORY)]: '.spec.domain.resources.requests.memory',
+  [getFieldId(VMSettingsField.CPU)]: '.spec.domain.cpu.cores',
+};
+
+export const getTemplateValidations = (
+  options: UpdateOptions,
+  fieldId: VMSettingsRenderableField,
+): CommonTemplatesValidation[][] => {
+  // Proceed if there are no validation for the specific fieldId
+  if (!(fieldId in IDToJsonPath)) {
+    return [];
+  }
+  const { getState, id } = options;
+  const state = getState();
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
+  // Get templates based on OS, WorkloadProfile and Flavor
+  const relevantTemplates = getRelevantTemplates(commonTemplates, os, workloadProfile, flavor);
+
+  // Get all the validations from the relevant templates
+  const validations = getValidationsFromTemplates(relevantTemplates);
+
+  // Return all the validations which are relevant for the field
+  return getFieldValidations(validations, IDToJsonPath[fieldId]);
+};
+
+export const getIntegerValidationType = (
+  validations: CommonTemplatesValidation[],
+): IntegerValidationObj => {
+  /*
+  Integer validation can be one of three options:
+  1. minimum: only 'min' rule | validation = [{min: 1}, ...]
+  2. maximum: only 'max' rule | validation = [{max: 10}, ...]
+  3. between: both 'min' and 'max' rules | validation = [{min: 1}, {max: 10}, ...] or [{min: 1, max: 10}, ...]
+  */
+  let integerValidationObj: IntegerValidationObj;
+  // eslint-disable-next-line consistent-return
+  validations.forEach((validation) => {
+    const validationAttrs = getValidationAttributes(validation);
+    if (validationAttrs.includes('min') && validationAttrs.includes('max')) {
+      return getBetweenIntegerValidationObj(
+        { min: validation.min, max: validation.max },
+        validation.message,
+      );
+    }
+    if (validationAttrs.includes('min')) {
+      if (integerValidationObj.type === IntegerValidtionType.max) {
+        // change the type to 'between' because we have two seperate validations of min and max
```

Comment:
hmm, when calling this method we should have only one rule - see comment https://github.com/openshift/console/pull/3839/files#r362532470

Reply:
not necessarily.
Most of the times we will have just one, it will be one of the three:
1. {min:1},{cpu:2},....
2. {max:10},{cpu:2},...
3.{min:1,max:10},{cpu:2},....
but we have here an edge case when a between validation can be written as a 2 separated validations in a single template:
4. {min:1},{max:10},{cpu:2},....

Reply:
this method should be called with one rule at a time, you should check why it is called with an array ...